services/custom-example/src/pages/index.tsx

```diff
@@ -129,12 +129,26 @@ const Home: NextPage = () => {
   const sendTx = async (e: FormEvent & { currentTarget: HTMLFormElement }) => {
     e.preventDefault();
     if (!inputMessage.trim()) return;
-    void writeContractAsync({
+
+    const userOperationHash = await writeContractAsync({
       address: config.privateErc20Address,
       abi: privateErc20Abi,
       functionName: 'mint',
       args: [parseEther(inputMessage.trim())],
     });
+
+    const userOpReceipt = await gianoConnector.waitForUserOperationReceipt(userOperationHash);
+    // Without wagmi connector, the provider can be used directly:
+    // const userOpReceipt = await gianoProvider.request({
+    //    method: 'getUserOperationReceipt',
+    //    params: [userOperationHash],
+    // });
+
+    if (userOpReceipt) {
+      console.log('✅ User operation receipt received:', userOpReceipt);
+      return;
+    }
+    console.log('❌ User operation receipt not found');
```

Comment:
This frontend behaviour is broken by the custom `submitUserOperation` injection; being able to wait for the receipt separately from the submission